File: database-driver-api/src/main/java/de/lino/database/database/entity/DatabaseEntry.java
package de.lino.database.database.entity;

import de.lino.database.json.JsonDocument;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * A single record stored inside a {@link de.lino.database.database.DatabaseSection}, consisting
 * of a unique primary key and its associated {@link JsonDocument} content.
 * <p>
 * By convention, the wrapped document (see the generated {@code getDocument()} accessor) stores
 * its actual payload under the {@code "data"} key, which can be retrieved directly via
 * {@link #getMetaData()}.
 */
@Getter
@RequiredArgsConstructor
public class DatabaseEntry {

    /**
     * The primary key that uniquely identifies this entry within its section.
     */
    @NotNull
    private final String id;

    /**
     * The full document backing this entry, including its {@code "data"} envelope.
     */
    @NotNull
    private final JsonDocument document;

    /**
     * Contains the MetaData of the entry
     * @return JsonDocument
     */
    @Nullable
    public JsonDocument getMetaData() {
        return this.document.getMetaData("data");
    }

}

File: python/database-driver-api/src/database_driver/api/database/entity/database_entry.py
"""Mirror of ``de.lino.database.database.entity.DatabaseEntry``."""

from __future__ import annotations

from dataclasses import dataclass

from database_driver.api.json.json_document import JsonDocument


@dataclass(frozen=True, slots=True)
class DatabaseEntry:
    """A single record stored inside a ``DatabaseSection``, consisting of a unique
    primary key and its associated ``JsonDocument`` content.

    By convention, the wrapped document stores its actual payload under the ``"data"``
    key, which can be retrieved directly via :meth:`get_meta_data`.

    Attributes:
        id: The primary key that uniquely identifies this entry within its section.
        document: The full document backing this entry, including its ``"data"`` envelope.
    """

    id: str
    document: JsonDocument

    def get_meta_data(self) -> JsonDocument | None:
        """Returns the entry's payload: the nested document under the ``"data"`` key, or
        ``None`` if the backing document has no such object."""
        return self.document.get_meta_data("data")
